downloader/src/main/kotlin/VersionControlSystem.kt

```diff
@@ -358,6 +358,11 @@ abstract class VersionControlSystem {
 
         addMetadataRevision(pkg.vcsProcessed.revision)
 
+        if (type == VcsType.GIT && pkg.vcsProcessed.revision == "master") {
+            // Also try with Git's upcoming default branch name in case the repository is already using it.
+            addMetadataRevision("main")
```

Comment:
Should we also verify that `listRemoteBranches()` contains "main" before adding it?

Reply:
That check basically is implicitly done as part of `addMetadataRevision()` when determining whether `main` is a fixed revision, which involves also listing the remote branches.

However, you're right that there's a general issue with the way `isFixedRevision()` currently works: For non-existing branches, it cannot *generally* tell whether it's a fixed revision or not. For example, consider the revision string "deadbeef". That could either be a non-existing branch name, or an existing short SHA1 for a commit. Distinguishing this better has been on my mental TODO-list for a while, but I'd like to do that separately of this PR.